File: backend/app/api/v1/endpoints/health.py
"""Health check API endpoints."""

import logging
from fastapi import APIRouter, Depends, status
from sqlalchemy.orm import Session
from sqlalchemy import text

from app.api.v1.deps import get_db
from app.schemas.common import HealthCheck
from app.core.config import settings

logger = logging.getLogger(__name__)

router = APIRouter(tags=["health"])


@router.get(
    "/health",
    response_model=HealthCheck,
    status_code=status.HTTP_200_OK,
    summary="Health check",
    description="Check if the service is healthy and running"
)
async def health_check() -> HealthCheck:
    """
    Basic health check endpoint.
    
    Returns:
        HealthCheck response with service status
    """
    return HealthCheck(
        status="healthy",
        version=settings.APP_VERSION,
        details={
            "app_name": settings.APP_NAME,
            "ai_provider": settings.AI_PROVIDER,
            "debug_mode": settings.DEBUG
        }
    )


@router.get(
    "/health/db",
    response_model=HealthCheck,
    status_code=status.HTTP_200_OK,
    summary="Database health check",
    description="Check database connectivity"
)
async def database_health_check(
    db: Session = Depends(get_db)
) -> HealthCheck:
    """
    Database connectivity health check.
    
    Args:
        db: Database session
        
    Returns:
        HealthCheck response with database status
    """
    try:
        # Execute a simple query to check database connectivity
        result = db.execute(text("SELECT 1"))
        result.scalar()
        
        return HealthCheck(
            status="healthy",
            version=settings.APP_VERSION,
            details={
                "database": "connected",
                "database_url": settings.DATABASE_URL.split("://")[0]  # Show only the DB type
            }
        )
        
    except Exception as e:
        logger.error(f"Database health check failed: {e}")
        return HealthCheck(
            status="unhealthy",
            version=settings.APP_VERSION,
            details={
                "database": "disconnected",
                "error": str(e)
            }
        )

File: backend-node/src/routes/health.js
import express from 'express';
import { testConnection } from '../database/index.js';

const router = express.Router();

// Health check endpoint
router.get('/', async (req, res) => {
  const dbHealthy = await testConnection();
  
  res.json({
    status: 'healthy',
    timestamp: new Date().toISOString(),
    version: process.env.APP_VERSION || '1.0.0',
    details: {
      app_name: process.env.APP_NAME || 'Cal AI Backend',
      ai_provider: process.env.AI_PROVIDER || 'anthropic',
      debug_mode: process.env.NODE_ENV === 'development',
      database: dbHealthy ? 'connected' : 'disconnected'
    }
  });
});

// Database health check
router.get('/db', async (req, res) => {
  const isHealthy = await testConnection();
  
  if (isHealthy) {
    res.json({
      status: 'healthy',
      message: 'Database connection is active'
    });
  } else {
    res.status(503).json({
      status: 'unhealthy',
      message: 'Database connection failed'
    });
  }
});

export default router;